src/pages/WithdrawSwiftPage/index.tsx

```diff
@@ -0,0 +1,306 @@
+  readonly profileStore = this.props.rootStore!.profileStore;
+  readonly socketStore = this.props.rootStore!.socketStore;
+
+  @observable operationId: string = '';
+
+  @computed
+  get balance() {
+    if (!this.walletStore.tradingWallets.length) {
+      return 0;
+    }
+
+    const {assetId} = this.props.match.params;
+    const balanceModel = this.walletStore.tradingWallets[0].balances.find(
+      (assetBalance: BalanceModel) => {
+        return assetBalance.assetId === assetId;
+      }
+    );
+
+    if (balanceModel) {
+      return balanceModel.balance;
+    }
+
+    return 0;
+  }
+
+  componentDidMount() {
+    window.scrollTo(0, 0);
+  }
+
+  render() {
+    const {assetId} = this.props.match.params;
+    const asset = this.assetStore.getById(assetId);
+    const requiredErrorMessage = (fieldName: string) =>
+      `Field ${fieldName} should not be empty`;
+
+    return (
+      <div>
+        <div className="container">
+          <div className="withdraw-swift">
+            <div className="withdraw-swift__title">Withdraw SWIFT</div>
+            <div className="withdraw-swift__subtitle">
+              {this.balance} {!!asset && asset!.name}
+              <div className="withdraw-swift__subtitle-label">Available</div>
+            </div>
+            <div className="withdraw-swift__description">
+              For the withdrawal of funds, the following account details will be
+              used.
+            </div>
+            <Formik
+              initialValues={this.withdrawStore.withdrawSwift}
+              enableReinitialize
+              validationSchema={Yup.object().shape({
+                accHolderAddress: Yup.string().required(
+                  requiredErrorMessage('Address of the accountholder')
+                ),
+                accHolderCity: Yup.string().required(
+                  requiredErrorMessage('City of the accountholder')
+                ),
+                accHolderZipCode: Yup.string().required(
+                  requiredErrorMessage('Zip code of the accountholder')
+                ),
+                accountName: Yup.string().required(
+                  requiredErrorMessage('Account Name')
+                ),
+                accountNumber: Yup.string().required(
+                  requiredErrorMessage('Account Number')
+                ),
+                amount: Yup.number()
+                  .moreThan(0, requiredErrorMessage('Amount'))
+                  .required(requiredErrorMessage('Amount')),
+                bankName: Yup.string().required(
+                  requiredErrorMessage('Bank Name')
+                ),
+                bic: Yup.string().required(requiredErrorMessage('BIC'))
+              })}
+              // tslint:disable-next-line:jsx-no-lambda
+              onSubmit={this.handleSubmit}
+              render={this.renderForm}
+            />
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  private listenSocket = async (formikBag: any) => {
```

Comment:
Can you split this function into smaller parts as you did before?

Reply:
Splitted